### ISSUE TITLE
"GitHub rate limit exceeded" when trying to push to a GitHub repo

### ISSUE THREAD

**Issue by microbit-carlos** [microsoft/pxt-microbit#5373]
I've got this message for the first time this morning (UK time). This was simply after opening MakeCode, importing a GitHub URL, making a few changes, and trying to push (at this point it asked me to login into GitHub, which I did). No other MakeCode/GitHub activity done before that, so very unlikely that I have been really hitting the rate limit.

I then waited over an hour and tried again, same error.

<img width="1729" alt="Screenshot 2023-10-18 at 12 26 16" src="https://github.com/microsoft/pxt-microbit/assets/29712657/c0c664a2-859d-4a19-9975-7127303110af">

Is the `GitHub rate limit exceeded, please wait and try again` message produced by MakeCode or the GitHub API actual response?
It's possible there is a different API error being accidentally identified as a rate limit?

**Comment by abchatra** [comment]
I can reproduce the same issue for Mirosoft/pxt-data-streamer repo. Looks like there is some additional permission we need for org level repo.

**Comment by martinwork** [comment]
Reported in support ticket https://support.microbit.org/helpdesk/tickets/75330 (private)

**Comment by abchatra** [comment]
@martinwork can you point to which repo they are trying to push this to? This is particular to org level repo like Microsoft.

**Comment by martinwork** [comment]
@abchatra I'll check with them.

**Comment by martinwork** [comment]
Yes, it seems it is about an organisation. It is OK when they fork to a personal account.
![rate limit exceeded](https://github.com/microsoft/pxt-microbit/assets/989126/30462594-8b85-4154-898e-6648b464e535)
